sdk/search/Azure.Search.Documents/src/Generated/SearchServiceModelFactory.cs

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Search.Documents.Indexes.Models;
+using Azure.Search.Documents.Models;
+
+namespace Azure.Search.Documents
+{
+    /// <summary> Model factory for SearchService read-only models. </summary>
+    public static partial class SearchServiceModelFactory
```

Comment:
Noob questions here -
1. What is this needed for?
2. Will I manually update this file if a change is needed (like yanking out some model type)?

Reply:
These classes are only for customer test - to create models that have members customers can't otherwise set because they are response-only, for example. See https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking for more details.

Reply:
Thanks Heath.